homeassistant/components/russound_rio/entity.py

```diff
@@ -0,0 +1,70 @@
+"""Base entity for Russound RIO integration."""
+
+from collections.abc import Awaitable, Callable, Coroutine
+from functools import wraps
+from typing import Any, Concatenate
+
+from aiorussound import Controller
+
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, RUSSOUND_RIO_EXCEPTIONS
+
+
+def command[_EntityT: RussoundBaseEntity, **_P](
+    func: Callable[Concatenate[_EntityT, _P], Awaitable[None]],
+) -> Callable[Concatenate[_EntityT, _P], Coroutine[Any, Any, None]]:
+    """Wrap async calls to raise on request error."""
+
+    @wraps(func)
+    async def decorator(self: _EntityT, *args: _P.args, **kwargs: _P.kwargs) -> None:
+        """Wrap all command methods."""
+        try:
+            await func(self, *args, **kwargs)
+        except RUSSOUND_RIO_EXCEPTIONS as exc:
+            raise HomeAssistantError(
+                f"Error executing {func.__name__} on entity {self.entity_id},"
+            ) from exc
+
+    return decorator
+
+
+class RussoundBaseEntity(Entity):
+    """Russound Base Entity."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        controller: Controller,
+    ) -> None:
+        """Initialize the entity."""
+        self._instance = controller.instance
+        self._controller = controller
+        self._primary_mac_address = (
+            controller.mac_address or controller.parent_controller.mac_address
+        )
+        self._device_identifier = (
```

Comment:
The `self._device_identifier` instance attribute only seems to be used in this method. Then it doesn't need to be an instance attribute but can just be a local variable. Alternatively since it's only used once, just inline the expression where it's needed.